vaadin-checkbox-flow-parent/vaadin-checkbox-testbench/src/main/java/com/vaadin/flow/component/checkbox/testbench/CheckboxGroupElement.java

```diff
@@ -0,0 +1,131 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.checkbox.testbench;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.vaadin.flow.component.checkbox.testbench.CheckboxElement;
+import com.vaadin.testbench.HasHelper;
+import com.vaadin.testbench.TestBenchElement;
+import com.vaadin.testbench.elementsbase.Element;
+
+@Element("vaadin-checkbox-group")
+public class CheckboxGroupElement extends TestBenchElement
+        implements HasHelper {
+
+    /**
+     * Gets a list of all available options.
+     *
+     * @return a list of the options (visible text)
+     */
+    public List<String> getOptions() {
+        return getCheckboxes().stream().map(CheckboxElement::getLabel)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets all checkboxes which are part of this group.
+     *
+     * @return a list of all checkboxes in this group
+     */
+    public List<CheckboxElement> getCheckboxes() {
+        return $(CheckboxElement.class).all();
+    }
+
+    /**
+     * Selects/unselects the first checkbox matching the given text.
+     *
+     * @param text
+     *            the text of the option to select
+     */
+    public void changeSelectionByText(String text) {
+        Optional<CheckboxElement> checkbox = getCheckboxByText(text);
+        if (!checkbox.isPresent()) {
+            throw new NoSuchElementException(
+                    "No item with text '" + text + "' found");
+        }
+
+        checkbox.get().setChecked(!checkbox.get().isChecked());
+    }
+
+    public List<String> getSelectedTexts() {
+        Stream<CheckboxElement> button = getSelectedCheckboxes();
+        return button.map(CheckboxElement::getLabel)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the selected checkboxes.
+     *
+     * @return an stream with the selected checkboxs elements.
+     */
+    private Stream<CheckboxElement> getSelectedCheckboxes() {
+        return getCheckboxByChecked(true);
+    }
+
+    /**
+     * Gets the checkbox which is part of this group and has the given visible
+     * text.
+     *
+     * @param text
+     *            the text to search for
+     * @return an optional with the checkbox element or an empty optional if no
+     *         match was found
+     */
+    private Optional<CheckboxElement> getCheckboxByText(String text) {
```

Comment:
We should also support `getCheckboxByValue(String value)` to be aligned with `RadioButtonElement`.
That would require adding `getValue()` to the `CheckboxElement`.

Reply:
Text here refers to label text, not a value of individual checkbox. 

If you are referring by value about getting e.g. selected checkboxes, with this API it goes like:

getCheckboxes().stream().filter(checkbox -> checkbox.isChecked())

Reply:
Right, somehow I missed that `getRadioButtonByValue` is a private method. Nevermind then.